### PR TITLE
fix: check user is-existed when add member

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
check user is-existed when add member

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda-org.erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb24iOlsxMDY5LDg4Myw4ODEsMTA5MCw3NzJdLCJtZW1iZXIiOlsiMTAwMTIwNSJdfQ%3D%3D&id=282243&iterationID=1090&pId=0&type=BUG)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  Fix the bug that empty user in dice_member            |
| 🇨🇳 中文    |  修复了项目成员列表存在空成员的bug            |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Effet** [comment]
add a db migration to fix dirty data?

**Comment by chengjoey** [comment]
> add a db migration to fix dirty data?

apply a erda issue ticket to delete dirty data

File: pkg/ucauth/user_admin.go [hunk 1]
```diff
@@ -342,7 +342,11 @@ func (c *UCClient) findUsersByQuery(query string, idOrder ...string) ([]User, er
 		}
 		var orderedUsers []User
 		for _, id := range idOrder {
-			orderedUsers = append(orderedUsers, userMap[id])
+			user, ok := userMap[id]
```
Comment (Effet) [inline pkg/ucauth/user_admin.go:345]: how to proven there exist some case `ok = false`?

**Review by Effet — CHANGES_REQUESTED** [review]

File: pkg/ucauth/user_admin.go (hunk 1 above)
Comment (chengjoey) [reply pkg/ucauth/user_admin.go:345]: For example, there are two user IDs, one exists in uc and the other does not exist. Using these two user IDs to query uc, only existing users will be returned. So the not-existed userID will `ok = false`

Comment (Effet) [reply pkg/ucauth/user_admin.go:345]: In which situation "the caller pass non-exist userID", as you say only the caller give use bad userID will cause this problem, and what if there are no bad-caller?

**Review by Effet — APPROVED** [review]

**Pull request merged**